### DISCUSSION TITLE
Meeting 2024-05-21

### DISCUSSION THREAD

**Discussion by jamesmunns** [rust-embedded/wg#758] [General]
This is the agenda for the meeting today. Please feel free to add items you'd like to add as comments here.

**Comment by Ben-PH** [comment]
I would like to discuss the possibility/logistics/plans/etc around adding some time-oriented traits. https://github.com/rust-embedded/embedded-hal/issues/602 is something I started work on today, and would like to get a read on how others feel obout the initiative.

**Reply by jamesmunns** [reply]
Lots of discussion, recommendation was to:

* get embedded-time and/or fugit maintained (or a fork that is maintained)
* Write an RFC for inclusion of traits in embedded-hal, particularly discussing design tradeoffs for app, HAL, and driver users
* Discuss in the RFC what does/doesn't make sense, follow up on any feedback, get consensus (or not).

**Reply by Ben-PH** [reply]
I also caught on that such a project sheuld be a `rust-embedded/emdedded-time` of some sort.

**Reply by Ben-PH** [reply]
- At a cursory glance, `fugit` is already adequately maintained. `esp-hal` is a well resourced project, and they depend on fugit for their time-tracking implementations
- embedded-time certainly needs revitalisation of some sort. I'm not aware of the context of this projects status, so can't give much comment there. I'll follow up when I have more to add.
- I've put in a significant, and perhaps controversial PR which (minor) version bumps fugit, and migrates const genericts to typenums. I don't expect this to go through, at least not without extensive engagement, but the story of its acceptance/rejection will provide context for the broader ecosystem.
- I start putting together the RFC today.

**Reply by Ben-PH** [reply]
RFC is up: https://github.com/rust-embedded/wg/pull/762

**Comment by sourcebox** [comment]
rust-embedded-community:
- what's missing for a 0.4 release of usb-device? There's an important fix for large config descriptors.
- can this be merged? https://github.com/rust-embedded-community/meta/pull/24

**Reply by ryan-summers** [reply]
Hi @sourcebox, I primarily handle `usb-device`. There's nothing blocking a 0.4 release except for the lengthy time it takes to propogate downstream into dependant crates. Because of that, I was holding off for a bit to see if anyone else had breaking API changes they wanted to propose before I released. I haven't seen any though, so I'll look into a release shortly.

**Comment by jamesmunns** [comment]
`cortex-m-rt` v0.7.4 yank and v0.7.5 release:

* https://github.com/rust-embedded/cortex-m/pull/535
* https://github.com/rust-embedded/cortex-m/pull/533

We also might need to merge this to fix CI first:

* https://github.com/rust-embedded/cortex-m/pull/534

**Reply by jamesmunns** [reply]
dion:

> Basically, the macros crate was updated but not released and the updated cmrt crate still depended on the old macros version. This caused the hardfault handler to give an unsound exception frame argument.
>
> If you use the exception frame, the cpu would hardfault again, which locks up the cpu

**Reply by jamesmunns** [reply]
jannic:

> And the missing link_section will cause a linker error for non-trivial examples if the trampoline is enabled.

**Reply by jamesmunns** [reply]
Outcome: get the @rust-embedded/cortex-m team to review and merge these PRs, release as soon as we are able to.

**Comment by BartMassey** [comment]
Short update on Discovery Book plans and status:

* As discussed previously, the Discovery Book needs some love.
* We discussed this pretty extensively at RustNL. There was a reasonable consensus there. There are notes from that around.
* I'm behind as usual. I'm writing an RFC for the initial phase as we speak. I'll get it posted ASAP and we can discuss next week.
* I have larger documentation plans in the works. Help will be wanted soon.

**Reply by jamesmunns** [reply]
Link to RFC: https://github.com/rust-embedded/wg/pull/759
